examples/authentication/server/src/token_exchanger.rs

```diff
@@ -0,0 +1,144 @@
+//
+// Copyright 2020 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+//! Utilities for exchanging authorisation codes for identity tokens and validating the resulting
+//! tokens.
+//!
+//! For reference, see:
+//! https://developers.google.com/identity/protocols/oauth2/openid-connect#exchangecode
+
+use jsonwebtoken::{decode, decode_header, Algorithm, DecodingKey, Header, Validation};
+use log::info;
+use serde::{Deserialize, Serialize};
+use std::{error, fmt};
+use url::form_urlencoded;
+
+use crate::http_client;
+
+/// Claims encoded into the identity token
+#[derive(Debug, Serialize, Deserialize)]
```

Comment:
Add link https://developers.google.com/identity/protocols/oauth2/openid-connect#an-id-tokens-payload

also perhaps add some comments to the fields, or use friendlier name and override the name of the serialized field using some serde annotation.

Reply:
Done.